### PR TITLE
[PLATFORM-1335]: Build for ARM

### PR DESCRIPTION
**Pull request by MaeIsBad**
https://prima-assicurazioni-spa.myjetbrains.com/youtrack/issue/PLATFORM-1335

I switched over to using cross-rs instead of a dockerfile, because cross compiling is hell, unless you use qemu, which would kill the performance.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -22,7 +22,7 @@ serde_json = "1.0.80"
 
 base64 = "0.13.0"
 base64-url = "1.4.13"
-openssl = "0.10.55"
+openssl = { version = "0.10.55", features = ["vendored"]}
```
Comment (cottinisimone) [inline Cargo.toml:25]: We removed it from paperboy :). Why is it needed here? Compatibility with arm?

Comment (MaeIsBad) [reply Cargo.toml:25]: Yeah finding a prebuilt openssl that can be statically linked to arm with musl is a nightmare

Comment (cottinisimone) [reply Cargo.toml:25]: Agreed!

Comment (MaeIsBad) [reply Cargo.toml]: Done

**Review by cpiemontese — APPROVED** [review]
:+1:

Really nice for `cross`, I "discovered" it while investigating cross compilation for `piccolo` and I was looking forward to see it in action

**Comment by cpiemontese** [comment]
Also: where does `buildah` come from (in the pipeline)?

**Comment by MaeIsBad** [comment]
It's installed as a part of the ubuntu-latest vm

https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md#tools

**Pull request merged**